coding-standards-and-naming.rst

```diff
@@ -332,76 +147,47 @@ the symbol names could change in line with these guidelines:
-data-structures like ``ArrayObject``, exceptions and iterators.
+The above guidelines recommend against the global ``FFI`` class used by the
+``ffi`` extension. Using ``FFI\FFI`` would be preferred.
 
-The distinction between these three extensions is somewhat murky from an
-end-user perspective, and largely historical. Symbols have moved between these
-extensions, e.g. the ``Iterator`` interface originated in spl, but now lives in
-Core.
+Existing Non-Namespaced Symbols and Consistency
+===============================================
 
-Because these extensions combine a lot of unrelated or only tangentially related
-functionality, symbols should not be namespaced under the ``Core``, ``Standard``
-or ``Spl`` namespaces. Instead, these extensions should be considered as a
-collection of different components, and should be namespaced according to these.
-
-For example, ``str_contains()`` could become ``Str\contains()``, ``fopen()``
-could become ``File\open()``, and ``password_hash()`` could become
-``Password\hash()``. (These are non-normative examples, the RFC does not propose
-using these specific namespaces.)
+When adding new symbols to existing extensions it is RECOMMENDED to be
+consistent with existing symbols, rather than to follow the namespacing
+guidelines.
 
-Existing non-namespaces symbols and consistency
------------------------------------------------
-
-When adding new symbols to existing extensions, it is more important to be
-consistent with existing symbols than to follow the namespacing guidelines.
-
-For example, the ``array_is_list()`` function added in PHP 8.1 should indeed be
-called ``array_is_list()`` and should not be introduced as ``Array\is_list()``
-or similar. Unless and until existing ``array_*()`` functions are aliased under
-an ``Array\*`` namespace, new additions should continue to be of the form
-``array_*()`` to maintain horizontal consistency.
+For example, the ``array_is_list()`` function added in PHP 8.1 MUST indeed be
+called ``array_is_list()`` and MUST NOT have be introduced as
+``Array\is_list()`` or similar. Unless and until existing ``array_*()``
+functions are aliased under an ``Array\*`` namespace, new additions MUST
+continue to be of the form ``array_*()`` to maintain horizontal consistency.
 
 This is a somewhat loose guideline, and applies more strongly to functions than
 classes. In particular, when new object-oriented elements are introduced into an
-extension that has historically been procedural, these may be namespaced. For
-example, if ``OpenSSLCertificate`` had only been introduced in PHP 8.1, it
-should have been named ``OpenSSL\Certificate``.
-
-For the Core/standard/spl extensions, the previous considerations on component
-subdivision apply. The fact that string and array functions are not namespaced
-does not preclude new namespaced components in these extensions.
+extension that has historically been procedural, these MAY be namespaced. For
```

Comment:
This says MAY, but the original document says should. I don't necessarily disagree, but it diverges from the original meaning.

Reply:
I was referring to:

> Instead, these extensions should be considered as a collection of different components, and **should** be namespaced according to these.

It's a bit unclear whether this "should" refers to this the "according to" part, or the namespacing itself, i.e. whether we prefer namespaces over a `category_` prefix.

Reply:
But you're right anyway, the sentence was already there. I missed that.